pkg/utils/gzip.go

```diff
@@ -0,0 +1,113 @@
+package utils
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+)
+
+// ExtractGzip extracts a gzipped archive to the specified directory and
+// returns the path to the root level of the extracted archive.
+// Caller must ensure that the destination directory exists.
+func ExtractGzip(file, destDir string) (root string, err error) {
```

Comment:
Good idea of moving this out of the tests so that we could reuse it; sounds like something we'll leverage frequently given the testing pattern that we've adopted!